src/components/StrCalcuLator.js

```diff
@@ -0,0 +1,65 @@
+import { errorConstans } from "../constants/index.js";
+
+export default class StrCalcuLator {
+  #inputStrs;
+  #separator;
+
+  constructor(inputStr) {
+    this.#inputStrs = this.#createStr(inputStr); // 커스텀 구분자 검증과 문자열 생성.
+    this.#inputStrs = this.#separate(this.#separator, this.#inputStrs); // 구분자로 분류후 배열로 생성.
+    this.#isNumber(this.#inputStrs); // 분리된 문자열에서 조건에 맞는지 검증
+  }
+
+  // 오류 검증 및 필요한 문자열로 변환하는 메서드
+  #createStr(inputStr) {
+    const separator = [":", ","];
+    let customBoolean = false;
+    // 커스텀 구분자를 가지는지 확인.
+    customBoolean = this.#isCustom(inputStr);
+    
+    if (customBoolean) { // 커스텀 구분자를 가졌을 때,
+      const tmpSep = inputStr.slice(2, 3); // 커스텀 구분자
+      inputStr = inputStr.slice(5,); // 커스텀 구분자를 제외한 문자열
+      separator.push(tmpSep);
+    }
+    this.#separator = separator;   
+    return inputStr;
+  }
+
+  #separate(separatorArr, strs) {
+    separatorArr.forEach((sep) => {
+      strs = strs.split(sep).join('');
+    })
+    strs = strs.split('').map((el) => +el);
+    return strs;
+  }
+
+  calculate() {
+    const sum = this.#inputStrs.reduce((acc, cur) => {
+      return acc + cur;
+    }, 0);
+    // 구분자를 이용해 문자열 구분.
+    return sum;
+  }
+  
+  // 입력문에서의 에러 상황 체크
+  // 1. 커스텀 구분자를 가지는지.
+  #isCustom = (inputStr) => {
+    const regExp = /(\/\/.{1}\\n){1}.{0,}/; 
+    const regExp2 = /(\/\/\D{1}\\n){1}.{0,}/;
```

Comment:
이 부분은 커스텀 구분자 선언부 내부에서
1. 한 글자 이상의 문자열을 갖을 것
2. 숫자는 포함하지 않을 것

이라는 조건이 걸려있는것으로 보이는데 (만약 틀렸다면 넘어가셔두 돼요!)

```suggestion
const regExp = /^\/\/(\D+?)\\n/
```
이렇게 통합하는건 어떨까요?

Reply:
커스텀 구분자를 가졌을 경우에, 따로 구분해주는 메서드를 실행시키려고 저런 방식을 쓰긴 했습니다.
구분 메서드 실행부분을 바꾸고, 이 부분은 통합해주는게 깔끔하긴 하겠네요